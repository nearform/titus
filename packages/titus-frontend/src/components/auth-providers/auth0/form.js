import React, { useEffect } from 'react'
import PropTypes from 'prop-types'

/**
 * This form will be used:
 * 1- when user is not authenticated yet, url is /login
 * 2- whn Auth0 invokes it as a callback, with authentication details. Url is /login?#access_token=...
 */
export const Form = ({ login, authentication }) => {
  useEffect(() => {
    authentication
      .parseHash()
      // in case the url contains details, trigger login to resume redirect on dashboard
      .then(isAuthenticated => isAuthenticated && login())
  })
  return (
    <button className="button" onClick={login}>
      Login through Auth0
    </button>
  )
}

Form.propTypes = {
  login: PropTypes.func.isRequired
}
