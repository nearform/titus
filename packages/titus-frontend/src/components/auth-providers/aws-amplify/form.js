import React from 'react'
import PropTypes from 'prop-types'

import LoginForm from '../../login-form'
import { loginFormSchema } from '../utils'

export const Form = ({ login, loginError }) => (
  <LoginForm
    login={login}
    schema={loginFormSchema}
    loginError={loginError}
    header={`Please provide AWS Cognito account details:`}
    allowChangePassword
  />
)

Form.propTypes = {
  login: PropTypes.func.isRequired,
  loginError: PropTypes.string
}

export default Form
