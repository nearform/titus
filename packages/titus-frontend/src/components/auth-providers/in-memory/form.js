import React from 'react'
import PropTypes from 'prop-types'
import LoginForm from '../../login-form'
import { loginFormSchema } from '../utils'

export const Form = ({ login, loginError }) => (
  <LoginForm
    login={login}
    schema={loginFormSchema}
    loginError={loginError}
    header={`Note: Any username followed by a password with at least four
        characters containing at least one letter or number will work.`}
  />
)

Form.propTypes = {
  login: PropTypes.func.isRequired,
  loginError: PropTypes.string
}

export default Form
