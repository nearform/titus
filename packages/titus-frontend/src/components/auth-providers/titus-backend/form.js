import React from 'react'
import PropTypes from 'prop-types'
import LoginForm from '../../login-form'
import { loginFormSchema } from '../utils'

export const Form = ({ login, loginError }) => (
  <LoginForm
    login={login}
    schema={loginFormSchema}
    loginError={loginError}
    header={`Please provide Auth0 account details:`}
  />
)

Form.propTypes = {
  login: PropTypes.func.isRequired,
  loginError: PropTypes.string
}

export default Form
