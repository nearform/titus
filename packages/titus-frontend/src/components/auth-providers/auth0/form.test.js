import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import { Form } from './form'

describe('<Form />', () => {
  afterEach(cleanup)

  it('should render correctly when authentication and login return false', async () => {
    const authentication = {
      parseHash: () => {
        return new Promise(resolve => {
          return resolve(false)
        })
      }
    }

    const { asFragment } = render(
      <Form
        login={() => {
          return false
        }}
        authentication={authentication}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly when authentication returns true and login returns false', async () => {
    const authentication = {
      parseHash: () => {
        return new Promise(resolve => {
          return resolve(true)
        })
      }
    }

    const { asFragment } = render(
      <Form
        login={() => {
          return false
        }}
        authentication={authentication}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly authentication and login return true', async () => {
    const authentication = {
      parseHash: () => {
        return new Promise(resolve => {
          return resolve(true)
        })
      }
    }

    const { asFragment } = render(
      <Form
        login={() => {
          return true
        }}
        authentication={authentication}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should trigger button correctly', async () => {
    const authentication = {
      parseHash: () => {
        return new Promise(resolve => {
          return resolve(true)
        })
      }
    }

    const mockFn = jest.fn()

    const { asFragment, getByText } = render(
      <Form login={mockFn} authentication={authentication} />
    )
    const button = getByText('Login through Auth0')
    fireEvent.click(button)

    expect(mockFn.mock.calls.length).toBe(1)
    expect(asFragment()).toMatchSnapshot()
  })
})
