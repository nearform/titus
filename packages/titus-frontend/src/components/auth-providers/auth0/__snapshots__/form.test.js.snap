// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render correctly authentication and login return true 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Login through Auth0
  </button>
</DocumentFragment>
`;

exports[`<Form /> should render correctly when authentication and login return false 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Login through Auth0
  </button>
</DocumentFragment>
`;

exports[`<Form /> should render correctly when authentication returns true and login returns false 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Login through Auth0
  </button>
</DocumentFragment>
`;

exports[`<Form /> should trigger button correctly 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Login through Auth0
  </button>
</DocumentFragment>
`;
